### PR TITLE
Using the timestamp in parentchain block as VC's issuance timestamp

### LINKED ISSUE #1282: Using the timestamp in parentchain block as VC's issuance timestamp
Currently the block number is used as VC's issuance timestamp. This needs to be changed to the timestamp in parentchain blocks.

### PR DESCRIPTION
**Pull request by outofxxx**
resolved #1282 

Three fields in VC content may affect the F/E:
1. issuanceBlockNumber   ->  issuanceTimestamp
2. expirationBlockNumber -> expirationTimestamp
4. createdBlockNumber     -> createdTimestamp

### CODE REVIEW

**Comment by Kailai-Wang** [comment]
May I know the background of this PR? It seems we don't need anything from the parachain, if we need timestamps, we can simply get it locally.

**Comment by outofxxx** [comment]
> 

As described in the issue, the `blocknumber` in the VC needs to be replaced with `timestamp`. The timestamp here can be from the parentchain or sidechain. This pr implements the timestamp extrinsic parsed from the parentchain block.

File: tee-worker/core/parentchain/indirect-calls-executor/src/lib.rs [hunk 1]
```diff
@@ -300,7 +300,7 @@ where
 			// Found VerifyIdentity extrinsic
 			let verify_identity = VerifyIdentity { block_number: parentchain_block_number };
 			// Found RequestVC extrinsic
-			let request_vc = RequestVC { block_number: parentchain_block_number };
+			let request_vc = RequestVC;
 			// Found BatchAll extrinsic
 			let batch_all = BatchAll { block_number: parentchain_block_number };
```
Comment (BillyWooo) [inline tee-worker/core/parentchain/indirect-calls-executor/src/lib.rs:305]: do we still block number here too?

File: tee-worker/litentry/core/credentials/src/lib.rs [hunk 2]
```diff
@@ -229,44 +229,33 @@ pub struct Credential {
...
+		debug!("generate credential from template, who: {:?}", account_id_to_string(&who),);
 
 		let mut vc: Self =
 			serde_json::from_str(s).map_err(|err| Error::ParseError(format!("{}", err)))?;
 		vc.issuer.mrenclave = shard.encode().to_base58();
 		vc.issuer.name = LITENTRY_ISSUER_NAME.to_string();
 		vc.credential_subject.id = account_id_to_string(who);
-		vc.issuance_block_number = bn;
-		vc.expiration_block_number = None;
+		vc.issuance_timestamp = now_as_millis();
```
Comment (BillyWooo) [inline tee-worker/litentry/core/credentials/src/lib.rs:246]: Underneath of `now_as_millis()`, there is SystemTime::now(). So will this part of code be running within Enclave? Is it still secure?

**Review by BillyWooo — COMMENTED** [review]
In general, I agreed with this approach to get local timestamp, instead of block number. Just worry about the security concern within one of my comments.

**Review by BillyWooo — COMMENTED** [review]
In general, I agreed with this approach to get local timestamp, instead of block number. Just worry about the security concern within one of my comments.

File: tee-worker/core/parentchain/indirect-calls-executor/src/lib.rs (hunk 1 above)
Comment (outofxxx) [reply tee-worker/core/parentchain/indirect-calls-executor/src/lib.rs:305]: No longer needed in `RequestVC`

File: tee-worker/litentry/core/credentials/src/lib.rs (hunk 2 above)
Comment (outofxxx) [reply tee-worker/litentry/core/credentials/src/lib.rs:246]: I think it is ok, sidechain also uses this method to generate timestamp

**Review by BillyWooo — APPROVED** [review]

**Comment by BillyWooo** [comment]
Please resolve the conflicts, then merge it. Thanks.

**Pull request merged**